integration-cli/docker_api_swarm_config_test.go

```diff
@@ -116,3 +116,100 @@ func (s *DockerSwarmSuite) TestAPISwarmConfigsUpdate(c *check.C) {
 	c.Assert(err, checker.IsNil, check.Commentf(string(out)))
 	c.Assert(status, checker.Equals, http.StatusBadRequest, check.Commentf("output: %q", string(out)))
 }
+
+func (s *DockerSwarmSuite) TestAPISwarmTemplatedConfig(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	referencedSecretSpec := swarm.SecretSpec{
+		Annotations: swarm.Annotations{
+			Name: "referencedsecret",
+		},
+		Data: []byte("this is a secret"),
+	}
+
+	referencedSecretID := d.CreateSecret(c, referencedSecretSpec)
+	c.Assert(referencedSecretID, checker.Not(checker.Equals), "", check.Commentf("secrets: %s", referencedSecretID))
+
+	referencedConfigSpec := swarm.ConfigSpec{
+		Annotations: swarm.Annotations{
+			Name: "referencedconfig",
+		},
+		Data: []byte("this is a config"),
+	}
+
+	referencedConfigID := d.CreateConfig(c, referencedConfigSpec)
+	c.Assert(referencedConfigID, checker.Not(checker.Equals), "", check.Commentf("configs: %s", referencedConfigID))
+
+	configSpec := swarm.ConfigSpec{
+		Annotations: swarm.Annotations{
+			Name: "templated_config",
+		},
+		Templating: &swarm.Driver{
+			Name: "golang",
+		},
+		Data: []byte("SERVICE_NAME={{.Service.Name}}\n" +
+			"{{secret \"referencedsecrettarget\"}}\n" +
```

Comment:
Updated the PR with new commits to store configs in the tmpfs secrets directory when they have secrets inserted inside them. Marking WIP because this depends on docker/swarmkit#2286.